tutorials/advanced/containers/container_test.py

```diff
@@ -10,17 +10,21 @@
 @rfm.simple_test
 class ContainerTest(rfm.RunOnlyRegressionTest):
     platform = parameter(['Sarus', 'Singularity'])
+    valid_systems = ['daint:gpu']
+    valid_prog_environs = ['builtin']
 
-    def __init__(self):
+    @rfm.run_after('init')
+    def set_container_variables(self):
         self.descr = f'Run commands inside a container using {self.platform}'
-        self.valid_systems = ['daint:gpu']
-        self.valid_prog_environs = ['builtin']
         image_prefix = 'docker://' if self.platform == 'Singularity' else ''
         self.container_platform = self.platform
         self.container_platform.image = f'{image_prefix}ubuntu:18.04'
         self.container_platform.command = (
             "bash -c 'cat /etc/os-release | tee /rfm_workdir/release.txt'"
         )
+
+    @rfm.run_before('sanity')
+    def set_sanity_patterns(self):
         os_release_pattern = r'18.04.\d+ LTS \(Bionic Beaver\)'
         self.sanity_patterns = sn.all([
             sn.assert_found(os_release_pattern, 'release.txt'),
```

Comment:
Actually, this one could go in the class body, couldn't it?

Reply:
If we ignore the following line, then yes. To me it seems that should be fine. @teojgo ?

Reply:
Yes you can move it to the class body.